src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { UpdateTodoArgs } from '../../types/UpdateTodoArgs';
+
+interface Props {
+  todo: Todo;
+  deleteTodo: (todoId: number) => void;
+  toggleTodoStatus:(
+    todoId: number,
+    args: UpdateTodoArgs
+  ) => void;
+  updatingTodosId: number[]
+  selectedTodoId: number | null;
+  setSelectedTodoId: (todoId: number | null) => void;
+  updateTodoTitle: (todoId: number, args: UpdateTodoArgs) => void;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  toggleTodoStatus,
+  updatingTodosId,
+  selectedTodoId,
+  setSelectedTodoId,
+  updateTodoTitle,
+
+}) => {
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+
+  const isUpdatingItem = updatingTodosId.includes(todo.id);
+
+  const handleToggleTodoStatus = () => toggleTodoStatus(
+    todo.id,
+    { completed: !todo.completed },
+  );
+
+  const clearSelectedTodoId = () => setSelectedTodoId(null);
+
+  const handleDoubleClick = (
+    event: React.MouseEvent<HTMLSpanElement, MouseEvent>,
+    todoId: number,
+  ) => {
+    event.preventDefault();
+    setSelectedTodoId(todoId);
+  };
+
+  const isSelectedTodoId = selectedTodoId === todo.id;
+
+  const handleInputChanges = (event:React.ChangeEvent<HTMLInputElement>) => (
+    setEditedTitle(event.target.value)
+  );
+
+  const changeTittleIfEdited = () => {
+    if (editedTitle !== todo.title) {
+      updateTodoTitle(todo.id, { title: editedTitle });
+    }
+  };
+
+  const handleOnBlur = () => {
+    changeTittleIfEdited();
+
+    clearSelectedTodoId();
+  };
+
+  const handleSubmit = (event:React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    changeTittleIfEdited();
+
+    clearSelectedTodoId();
+  };
+
+  const handleCancelEditing = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Escape') {
+      clearSelectedTodoId();
+    }
```

Comment:
When I add some words to the title, press ESC and reopen edit input - I this new words in edit input
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/5cf87fbb-8f8d-4943-a0ec-03b3742c2efc)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/3165fc04-8a9e-4ce9-8049-6903961188b5)